src/Claymate.tsx

```diff
@@ -46,16 +46,15 @@ const Claymate: React.FC<Props> = ({
     const index = scenes.findIndex((sc) => sc.id === id);
     if (index >= 0) {
       const remainingScenes = scenes.length - 1;
-      let newIndex;
       if (remainingScenes > 0) {
-        newIndex = index < remainingScenes ? index : remainingScenes - 1;
+        const newIndex = index < remainingScenes ? index : remainingScenes - 1;
+        const drawingSourceIndex =
+          index < remainingScenes ? index + 1 : index - 1;
+        updateScenes((prev: Scene[]) => prev.filter((item) => item.id !== id), {
+          index: newIndex,
+          drawing: scenes[drawingSourceIndex].drawing,
```

Comment:
I'm not sure if I understand why we need two index numbers.

Reply:
The source of the new scene after deletion is not necessarily based on the index that will be current after the deletion

Reply:
> The source of the new scene after deletion is not necessarily based on the index that will be current after the deletion

Oh, I see, sure. That was my misunderstanding. Thanks for explanation.